### DISCUSSION TITLE
Special "Mixer": Scanner output based on audio channels (not RF channels)

### DISCUSSION THREAD

**Discussion by thowe-switzerland** [rtl-airband/RTLSDR-Airband#251] [General]
I receive with one SDR (RSP1A) only on a bandwidth of 8MHz several ATC frequencies of an airport. I can record and stream these output channels individually in a very good quality.

But especially in times with little traffic, I want to monitor the channels together. With the audio channel mixer I can do this. But if two or more channels are active at the same time, there is only a murmur that is no longer understandable.

What I would like is a kind of "scanning audio mixer": make one output channel out of multiple input audio channels by selecting one channel only. I.e. if one of these input channels is active, only this active channel is output until this channel has been inactive for approx. 2 seconds. Then all channels are monitored again for activity. Optionally, one of the channels could be marked as priority. This channel could then immediately displace another active channel of no priority.

I don't see any solution within RTLSDR-Airband at the moment. Or did I miss something? Maybe there would be such a function in a third party tool for e.g. Icecast streams?

I would be very interested to hear what you think about this and if you see a solution.

Many thanks & best regards
Thomas

**Comment by charlie-foxtrot** [comment]
Are you asking about putting rtl-airband in scanning mode so only one channel is picked up at a time?  If so, see https://github.com/szpajder/RTLSDR-Airband/wiki/Configuring-channels-for-scan-mode

Or are you asking to monitor all channels at once but to put an audio buffer behind each channel and before the mixer such that all audio is captured and buffered until the other channels are quiet?  If you don't care about delay, this could be achieved by post-processing by enabling`split_on_transmission` on each channel then writing a script to watch the output directory and stitch together `.mp3` files.

Alternatively, have you tried using the mixer `balance` to send different channels to the left vs right (`1.0`, `-1.0`) ear so that it's easier to make out whats being said when multiple channels are transmitting?

**Reply by thowe-switzerland** [reply]
Thank you very much for your advice.

I have been using the scanning mode on the radio frequencies for years to realize a scanner with feed to LiveATC. But since I need all 4 ATC frequencies unscanned for another use case (research project), I have at the moment a highly redundant setup with several SDR: The broadband RSP1A receives all 4 frequencies simultaneously and produces from them in parallel a separate audio channel each, which are all recorded and streamed separately. At the same time there is the additional normal scanner with an Airspy mini, which additionally hops or scans over all frequencies for the combined LiveATC stream. This setup with two receivers works, but requires an antenna splitter and to compensate its loss an LNA between splitter and antenna. And needs more power and generates heat that has to be brought out of the case.

In principle, however, the RSP1A alone already receives everything it needs to simulate an additional channel in scanning mode from the four individual channels. Instead of frequency hopping simply directly by evaluating the 4 single audio channels in a special mixer with logic, which always selects only one channel with activity and mutes the others in the meantime.

What I want is exactly the same result as the RF scanning mode, except that the scan should be based on the four separate demodulated audio channels.

The positioning with the mixer in the stereo area between left and right, which you also mentioned, is also known to me. But I find it very tiring to have activity on multiple frequencies at the same time. That may perhaps go better with stereo headphones. But if you want to listen to the ATC stream from a speaker (e.g. smartphone), it becomes more difficult. For me, the scanning mode works better. Even if I miss something on another channel.

If my C/C++ knowledge wasn't so rusty, I would try to implement such a scanning mixer myself. But I still have the hope that it already exists. :-)
